core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -179,7 +180,11 @@ class HistoryServer(
   }
```

Comment:
@srowen I have deleted getApplicationListView() and now it directly uses iterator to do the filter and map. Let me know if you have any concern. Thanks!